scripts/tests/unit_tests.sh

```diff
@@ -45,25 +77,48 @@ utest_compile_unit || die "Failed"
 print Y "Running the unit-tests..."
 utest_run_unit || die "Failed"
 
-printn Y "Cleaning the existing project... "
+if [[ "$GRCOV_FILENAME" ]]; then
+  printn Y "Generating temp coverage file for unit tests..."  
+  _grcov_unit unit_"$GRCOV_FILENAME"
+fi
+
+printn Y "Cleaning the existing unit project... "
 utest_cleanup_unit || die "Failed"
 
 print Y "Compile the auth-unit-tests..."
 utest_compile_auth || die "Failed"
 
-print Y "Running the unit-tests..."
+print Y "Running the auth unit-tests..."
 utest_run_auth || die "Failed"
 
-printn Y "Cleaning the existing project... "
+if [[ "$GRCOV_FILENAME" ]]; then
+  printn Y "Generating temp coverage file for auth tests...$GRCOV_FILENAME"  
+  _grcov_unit auth_"$GRCOV_FILENAME"
+fi
+
+printn Y "Cleaning the existing auth project... "
 utest_cleanup_auth || die "Failed"
 
 print Y "Compile the native-messaging-unit-tests..."
 utest_compile_nativemessaging || die "Failed"
 
-print Y "Running the unit-tests..."
+print Y "Running the native messaging unit-tests..."
 utest_run_nativemessaging || die "Failed"
 
-printn Y "Cleaning the existing project... "
+if [[ "$GRCOV_FILENAME" ]]; then
+  printn Y "Generating temp coverage file for native messaging unit tests..."  
+  _grcov_unit nativemessaging_"$GRCOV_FILENAME"
+fi
+
+printn Y "Cleaning the existing native messaging project... "
 utest_cleanup_nativemessaging || die "Failed"
 
+if [[ "$GRCOV_FILENAME" ]]; then  
+  printn Y "merging temp files to finallcov.info... "
+  grcov -t lcov -o unit_final_lcov.info unit_"$GRCOV_FILENAME" auth_"$GRCOV_FILENAME" nativemessaging_"$GRCOV_FILENAME" || die "merging temp files to final failed"
```

Comment:
`"$GRCOV_FILENAME"` vs `unit_final_lcov.info`

Reply:
i'll rename it to take the given name from the yml file